### ISSUE TITLE
analysis: train tool-count proxy model and back-estimate trailer-only runs

### ISSUE THREAD

**Issue by petar-djukic** [petar-djukic/go-unix-utils#5029]
## Problem

Only ~200 of ~2,300 tasks have transcript-level tool counts (run 43 + recovered Feb-Mar). The other ~2,100 (the middle 6 weeks of runs, March 7 – April 18) have only commit trailers. To produce cross-run tool-call charts that span all runs, we need to estimate tool counts from trailers using a model trained on the labeled subset.

## Approach

Regression model:

```
tool_calls_total ~ tokens_input + tokens_cache_read + tokens_cache_creation + duration_seconds + loc_prod_delta + loc_test_delta + insertions + deletions + is_contract + target_kind
```

Train on the join of `tools.csv` (aggregated per task) with `tasks.csv`. Apply to unlabeled rows. Emit prediction intervals.

## Implementation

Depends on issues 5017 (`tasks.csv`) and 5020 (`tools.csv`, `task_turns.csv`).

Add to pixi: `pixi add scikit-learn`.

Create `analysis/scripts/calibrate_tool_proxy.py`:

Step 1: Build labeled set
- `tools_per_task = tools.csv.groupby('task_id').size()` → actual tool calls per task
- Also derive `bash_calls`, `go_build_calls`, `go_vet_calls`, `go_test_calls`, `read_calls`, `edit_calls`, `write_calls` per task
- Join with `tasks.csv` on `task_id` to get features

Step 2: Train regression
- Split 80/20 train/test, stratified by run (so runs are not split across folds)
- Train multiple models on `total_tool_calls` and on each subcategory (bash, go_build, etc.) separately:
  - Linear regression baseline
  - Random forest regressor
  - Gradient boosting regressor
- Report MAE, RMSE, R² for each model on test set.
- Pick best model by test R² for each target.

Step 3: Apply to unlabeled rows
- For each row in `tasks.csv` not in the labeled set, predict `est_total_tool_calls`, `est_bash_calls`, `est_go_build_calls`, etc.
- Compute 90% prediction interval (use quantile regression or bootstrap of the chosen model).
- Write `tasks_with_estimated_tools.csv` covering all ~2,300 tasks with new columns:
  - `est_total_tool_calls`, `est_total_tool_calls_lo`, `est_total_tool_calls_hi`
  - same for each subcategory
  - `is_predicted` (false for labeled rows, true for proxied rows)

Charts (under `analysis/charts/proxy/`):

1. **predicted_vs_actual_scatter.png** — train/test scatter for the chosen total_tool_calls model. 45-degree reference line. Title shows test R² and MAE.
2. **feature_importance.png** — bar chart of feature importances from random forest (only for total_tool_calls).
3. **estimated_tool_calls_trend.png** — line chart, x = run order (chronological), y = mean predicted tool calls per task per run. Shaded band shows 90% interval. Annotate "labeled" runs (transcript data exists) vs "proxied" runs.
4. **estimated_go_build_failures_trend.png** — same but for `est_go_build_failed_calls` (build the failed-call model from `tools.csv` filtering on `is_error=true`).

Script outputs `analysis/datasets/proxy_model_metrics.yaml`:

```yaml
total_tool_calls:
  best_model: gradient_boosting
  test_r2: 0.71
  test_mae: 4.3
  feature_importance:
    duration_seconds: 0.32
    tokens_output: 0.21
    loc_prod_delta: 0.18
go_build_calls:
  test_r2: 0.55
  test_mae: 1.2
training_set_size: 198
total_proxied_rows: 2109
```

## Observed In

Datasets from issues 5017 and 5020.

## Test Plan

- Test-set MAE < 8 tool calls for `total_tool_calls` (reasonable goal for ~200 training examples).
- Output `tasks_with_estimated_tools.csv` covers all rows in `tasks.csv` (no missing predictions for any task).
- For labeled rows, `is_predicted=false` and the actual count is preserved.
- For predicted rows, `est_total_tool_calls_lo <= est_total_tool_calls <= est_total_tool_calls_hi`.
- Feature importance chart should show `tokens_output` and `duration_seconds` among top features (intuitive: more output and more time → more tool calls).
- Trend chart should make visible whether tool calls per task are increasing or decreasing over the 92 runs.

**Comment by petar-djukic** [comment]
tokens: ~100k
